packages/interactivity/src/directives.js

```diff
@@ -205,21 +205,28 @@ const cssStringToObject = ( val ) => {
  * @param {string} type 'window' or 'document'
  * @return {void}
  */
-const getGlobalEventDirective =
-	( type ) =>
-	( { directives, evaluate } ) => {
+const getGlobalEventDirective = ( type ) => {
+	return ( { directives, evaluate } ) => {
+		const events = new Map();
 		directives[ `on-${ type }` ]
 			.filter( ( { suffix } ) => suffix !== 'default' )
 			.forEach( ( entry ) => {
+				const event = entry.suffix.split( '--' )[ 0 ];
```

Comment:
Can `entry.suffix` ever be `""` or `"--…"`? That may be something to improve in some earlier processing. We shouldn't be attaching event listeners to events like "".

Reply:
Suffix initially cannot be `""` . If you add a directive like `data-wp-on-document----a`, it will an empty string as an event, as you mention.
`data-wp-on-document--` directly crashes the Interactivity runtime.

Reply:
Thanks for opening #61249, let's use that to make sure only "good" suffixes reach this point.

[Do we want to use a limit to avoid splitting more than we'd use?
](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/split#limit)
```suggestion
				const event = entry.suffix.split( '--', 1 )[ 0 ];
```

Reply:
+1 to only split the suffix once.

Reply:
Done in https://github.com/WordPress/gutenberg/pull/61009/commits/d57523d436c0283ecde04c02a3ceade833edb682